File: PSP/graphql/backend/src/main/java/com/example/demo/data/repositories/PoiRepository.java
package com.example.demo.data.repositories;

import com.example.demo.data.modelo.PoiEntity;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.ListCrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Arrays;
import java.util.List;

@Repository
public interface PoiRepository extends ListCrudRepository<PoiEntity, Long> {
    @EntityGraph(attributePaths = {"ciudad"})
    List<PoiEntity> findAllByCiudad_id(Long idCiudad);

    @EntityGraph(attributePaths = {"ciudad"})
    List<PoiEntity> findAllByVisitas_Id(Long id);


    @Query("""
             SELECT p FROM PoiEntity p
             JOIN FETCH p.ciudad c
             """)
    List<PoiEntity> findAllWithCiudad();
}

File: PSP/graphql/frontend/src/main/kotlin/org/example/frontend/data/PoiRepository.kt
package org.example.frontend.data

import com.apollographql.apollo3.ApolloClient
import org.example.visitas.GetPoisQuery
import org.springframework.stereotype.Repository

@Repository
class PoiRepository(
    var apolloClient: ApolloClient

) {

    suspend fun getPOIs() = apolloClient.query(GetPoisQuery())
        .execute().data?.getPois
}
